lib/checkPullRequestTemplate.js

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Handler to check that the pull request template is followed.
+ */
+
+const { swapCharInPlace } = require('./utils');
+const KARMA_LINTER = 'linter/karma';
+
+/**
+ * Checks that the pull request's body contains the explanation.
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkPRExplanation = (pullRequest) => {
+  // Restrict search to the overview section.
+  const overviewText = '## Overview';
+  const overviewHeadingIndex = pullRequest.body.indexOf(overviewText);
+
+  // Checks that there is an overview.
+  if (overviewHeadingIndex !== -1) {
+    // Get the index of the next heading after the overview.
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      overviewHeadingIndex + overviewText.length
+    );
+
+    const overviewSection = pullRequest.body.substring(
+      overviewHeadingIndex,
+      nextHeadingIndex
+    );
+
+    let explanation = overviewSection.split('\n').find((line) => {
+      return line.startsWith('2');
```

Comment:
Add reasoning/examples behind each check - for increased readability.

Reply:
Done.